src/plugins/gmx/gmxplugin.cpp

```diff
@@ -0,0 +1,185 @@
+
+    stream.setAutoFormatting(true);
+    stream.setAutoFormattingIndent(2);
+
+    stream.writeComment("This Document is generated by GameMaker, if you edit it by hand then you do so at your own risk!");
+
+    stream.writeStartElement("room");
+    stream.writeStartElement("caption");
+    stream.writeEndElement();
+
+    stream.writeTextElement("width", QString::number(map->tileWidth() * map->width()));
+
+    stream.writeTextElement("height", QString::number(map->tileHeight() * map->height()));
+
+    stream.writeTextElement("vsnap", QString::number(map->tileHeight()));
+    stream.writeTextElement("hsnap", QString::number(map->tileWidth()));
+
+    stream.writeTextElement("isometric", QString::number((map->orientation() == Map::Orientation::Isometric ? 1 : 0)));
+
+    stream.writeStartElement("instances");
+
+    uint objectId = 0u;
+    foreach (const Layer *layer, map->layers()) {
+
+        if (layer->layerType() != Layer::ObjectGroupType) continue;
+        const ObjectGroup *objectLayer = static_cast<const ObjectGroup*>(layer);
+
+        foreach(const MapObject *object, objectLayer->objects()) {
+            stream.writeStartElement("instance");
+
+            stream.writeAttribute("objName", object->name());
+            stream.writeAttribute("x", QString::number((int)(object->x())));
+            stream.writeAttribute("y", QString::number((int)(object->y() - object->height())));
+
+            stream.writeAttribute("id", QString::number(++objectId));
+
+            stream.writeAttribute("locked", QString::number(0));
+            stream.writeAttribute("scaleX", QString::number(1));
+            stream.writeAttribute("scaleY", QString::number(1));
+
+            stream.writeEndElement();
+        }
+
+        if (file.error() != QFile::NoError) {
+            mError = file.errorString();
+            return false;
+        }
+
+    }
+
+    stream.writeEndElement();
+
+
+    stream.writeStartElement("tiles");
+
+
+
+    uint tileId = 0u;
+    int currentLayer = map->layers().size();
+
+    foreach (const Layer *layer, map->layers()) {
+
+        if (layer->layerType() != Layer::TileLayerType) continue;
+
+        const TileLayer *tileLayer = static_cast<const TileLayer*>(layer);
+        for (int y = 0; y < tileLayer->height(); ++y) {
+            for (int x = 0; x < tileLayer->width(); ++x) {
+                const Cell &cell = tileLayer->cellAt(x, y);
+                const Tile *tile = cell.tile;
+
+                if(tile) {
+                    stream.writeStartElement("tile");
+
+                    stream.writeAttribute("bgName", tile->tileset()->name());
+                    stream.writeAttribute("x", QString::number((int)(x * map->tileWidth())));
+                    stream.writeAttribute("y", QString::number((int)(y * map->tileHeight())));
+                    stream.writeAttribute("w", QString::number(map->tileWidth()));
+                    stream.writeAttribute("h", QString::number(map->tileHeight()));
```

Comment:
The tiles do not need to be the same size as the map tile size, so you should use `tile->size()` here.